dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/fileresource/FileResourceEventListener.java

```diff
@@ -78,19 +78,21 @@ public void save(FileSavedEvent fileSavedEvent) {
 
   @TransactionalEventListener
   @Async
-  public void saveImageFile(ImageFileSavedEvent imageFileSavedEvent) {
+  public void saveImageFile(ImageFileSavedEvent imageFileSavedEvent) throws NotFoundException {
     DateTime startTime = DateTime.now();
 
-    Map<ImageFileDimension, File> imageFiles = imageFileSavedEvent.getImageFiles();
+    Map<ImageFileDimension, File> imageFiles = imageFileSavedEvent.imageFiles();
 
     FileResource fileResource =
-        fileResourceService.getFileResource(imageFileSavedEvent.getFileResource());
+        fileResourceService.getFileResource(imageFileSavedEvent.fileResource());
 
     String storageId = fileResourceContentStore.saveFileResourceContent(fileResource, imageFiles);
 
     if (storageId != null) {
       fileResource.setHasMultipleStorageFiles(true);
 
+      authenticationService.obtainAuthentication(imageFileSavedEvent.userUid());
```

Comment:
To not leak the user context in the worker thread this should use a try-finally where in the finally you call the clear on the authenticationService.